src/streamlink/plugins/twitch.py

```diff
@@ -137,18 +137,44 @@
 
 Segment = namedtuple("Segment", "uri duration title key discontinuity scte35 byterange date map")
 
+LOW_LATENCY_MAX_LIVE_EDGE = 2
+
+
+def parse_condition(attr):
+    def wrapper(func):
+        def method(self, *args, **kwargs):
+            if hasattr(self.stream, attr) and getattr(self.stream, attr, False):
+                func(self, *args, **kwargs)
+        return method
+    return wrapper
+
 
 class TwitchM3U8Parser(M3U8Parser):
+    def __init__(self, base_uri=None, stream=None, **kwargs):
+        M3U8Parser.__init__(self, base_uri, **kwargs)
+        self.stream = stream
+
+    @parse_condition("disable_ads")
     def parse_tag_ext_x_scte35_out(self, value):
         self.state["scte35"] = True
 
     # unsure if this gets used by Twitch
+    @parse_condition("disable_ads")
     def parse_tag_ext_x_scte35_out_cont(self, value):
         self.state["scte35"] = True
 
+    @parse_condition("disable_ads")
     def parse_tag_ext_x_scte35_in(self, value):
         self.state["scte35"] = False
 
+    @parse_condition("low_latency")
+    def parse_tag_ext_x_twitch_prefetch(self, value):
+        segments = self.m3u8.segments
+        if segments:
+            last_segment = segments[-1:][0]
```

Comment:
Did I miss something or is the same as `segments[-1]`?

Reply:
This line was copied from back-to's low latency test plugin here:
https://gist.github.com/back-to/8e9ed3c60e5932d8c7a67ccd43b906d0/#file-twitch-py-L444
I didn't change it when I created this branch, because I wasn't sure if there was something going on with `__getitem__` or so. Looks like it's a simple list object. Do you want me to change it or can we leave it?

By the way, I'm also still not sure how the cloned prefetch segments will work in combination with the ads filtering, because if hls-live-edge is set to 1 or 2, which is now a requirement here, then regular segments won't be used anymore at all and you're re-using the metadata of the last regular segment in the playlist for all the prefetched ones. So if there's an ad break or the end of an ad break, then you will only notice it if the playlist gets refreshed quickly enough and the last regular segment has the metadata set accordingly. It does get refreshed though after each segment now, as mentioned above, so it's *probably* fine.

Reply:
Shouldn't you just be able to use `get_segment` to create the "prefetch" segments?

Reply:
@back-to what's the deal with `segments[-1:][0]`? Am I missing something important?

Reply:
> Am I missing something important?

nope